pkg/util/k8sutil/backup.go

```diff
@@ -14,244 +14,5 @@
 
 package k8sutil
 
-import (
-	"fmt"
-	"path"
-	"time"
-
-	"github.com/coreos/etcd-operator/pkg/util/constants"
-	"github.com/coreos/etcd-operator/pkg/util/retryutil"
-
-	appsv1beta1 "k8s.io/api/apps/v1beta1"
-	"k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/resource"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/util/intstr"
-	"k8s.io/client-go/kubernetes"
-)
-
-const (
-	BackupPodSelectorAppField = "etcd_backup_tool"
-	backupPVVolName           = "etcd-backup-storage"
-	fromDirMountDir           = "/mnt/backup/from"
-
-	PVBackupV1 = "v1" // TODO: refactor and combine this with pkg/backup.PVBackupV1
-)
-
-func CreateAndWaitPVC(kubecli kubernetes.Interface, clusterName, ns, storageClass string, volumeSizeInMB int) error {
-	name := makePVCName(clusterName)
-	claim := &v1.PersistentVolumeClaim{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:   name,
-			Labels: LabelsForCluster(clusterName),
-			Annotations: map[string]string{
-				"volume.beta.kubernetes.io/storage-class": storageClass,
-			},
-		},
-		Spec: v1.PersistentVolumeClaimSpec{
-			AccessModes: []v1.PersistentVolumeAccessMode{
-				v1.ReadWriteOnce,
-			},
-			Resources: v1.ResourceRequirements{
-				Requests: v1.ResourceList{
-					v1.ResourceStorage: resource.MustParse(fmt.Sprintf("%dMi", volumeSizeInMB)),
-				},
-			},
-		},
-	}
-	_, err := kubecli.CoreV1().PersistentVolumeClaims(ns).Create(claim)
-	if err != nil {
-		return err
-	}
-
-	// TODO: We set timeout to 60s here since PVC binding could take up to 60s for GCE/PD. See https://github.com/kubernetes/kubernetes/issues/40972 .
-	//       Change the wait time once there are official p99 SLA.
-	err = retryutil.Retry(4*time.Second, 15, func() (bool, error) {
-		var err error
-		claim, err = kubecli.CoreV1().PersistentVolumeClaims(ns).Get(name, metav1.GetOptions{})
-		if err != nil {
-			return false, err
-		}
-		if claim.Status.Phase != v1.ClaimBound {
-			return false, nil
-		}
-		return true, nil
-	})
-	if err != nil {
-		wErr := fmt.Errorf("fail to wait PVC (%s) '(%v)/Bound': %v", name, claim.Status.Phase, err)
-		return wErr
-	}
-
-	return nil
-}
-
+// What's this?
 var BackupImage = "quay.io/coreos/etcd-operator:latest"
```

Comment:
I think this was the image used for the backup sidecar pod. So we probably don't need this as well.

Reply:
Yep. Just delete it.

Reply:
@hasbro17 it is used here https://github.com/coreos/etcd-operator/blob/master/hack/build/operator/build#L27

Reply:
That's the build flag used for backup sidecar previously.
Delete that as well

Reply:
done. PTAL @hongchaodeng @hasbro17